tests/e2e/matmul/CMakeLists.txt

```diff
@@ -10,6 +10,33 @@
 
 iree_add_all_subdirs()
 
+iree_generated_e2e_runner_test(
+  NAME
+    e2e_matmul_cpu_arm_sme_nondt_f32_small_transpose_lhs
+  TEST_TYPE
+    matmul
+  GENERATOR
+    "generate_e2e_matmul_tests.py"
+  GENERATOR_ARGS
+    "--lhs_rhs_type=f32"
+    "--acc_type=f32"
+    "--shapes=small"
+  TEST_RUNNER
+    iree_tools_testing_e2e_iree-e2e-matmul-test
+  TARGET_BACKENDS
+    "llvm-cpu"
+  DRIVERS
+    "local-task"
+  COMPILER_FLAGS
+    "--iree-opt-data-tiling=false"
+    "--iree-llvmcpu-enable-scalable-vectorization"
+    "--iree-preprocessing-pass-pipeline=builtin.module\(util.func\(iree-preprocessing-transpose-matmul-pass{input=lhs}\)\)"
```

Comment:
What are the eventual goals for `TransposeMatmulPass`? I'm concerned about these sorts of verbose "preprocessing pass pipelines" making their way into test configurations and scripts unconstrained. We shouldn't require users (e.g. from the Python bindings) to spell out these explicit compiler passes.

Is this on a path to having its own top level option? Is it something that can be enabled unconditionally once more testing is done? Is it something that the compiler can decide when to enable on its own?

Reply:
The transposition done here is a short cut. Ideally this should all just fold into data-tiling. This is not something we want to enable by default in the compiler. This is kind of an opt-in transformation at the global level for a specific target hardware.
I would really think it would be better if this was a test downstream. Maybe a good structuring for the ARM/SVE/SME work is to use a plugin where you can have such opinionated transformations on by default and tested in the plugin like we do for the iree-amd-aie plugin.

Having said that, this test is only being run on ARM, so I am fine with it, but I think Scott has the final say here.

Reply:
>The transposition done here is a short cut. Ideally this should all just fold into data-tiling. This is not something we want to enable by default in the compiler. This is kind of an opt-in transformation at the global level for a specific target hardware.

+1 the preprocessing pass gives us a big win for minimal effort. As mentioned in [1] we'll be looking at alternatives such as data-tiling going forward.

>I would really think it would be better if this was a test downstream. Maybe a good structuring for the ARM/SVE/SME work is to use a plugin where you can have such opinionated transformations on by default and tested in the plugin like we do for the iree-amd-aie plugin.

We have no downstream.

[1] https://github.com/iree-org/iree/pull/17157#discussion_r1580541805